### DISCUSSION TITLE
Meeting 2024-07-23

### DISCUSSION THREAD

**Discussion by adamgreig** [rust-embedded/wg#778] [General]
This is the agenda for today's meeting. Please add anything you'd like to discuss below!

* Meetings: Tuesday 8pm Europe/Berlin time
* [Join the Matrix chat](https://matrix.to/#/#rust-embedded:matrix.org)
* [Nominated issues](https://github.com/search?q=org%3Arust-embedded+label%3Anominated+is%3Aopen&type=Issues)
* [IRC/Matrix logs](https://libera.irclog.whitequark.org/rust-embedded/2024-07-23)

**This meeting is primarily to discuss project goals per https://github.com/rust-embedded/wg/issues/775**

### Announcements

* None yet

**Comment by jamesmunns** [comment]
Regarding the goal process, I highly recommend reading [How the goal process works](https://rust-lang.github.io/rust-project-goals/2024h2/index.html#how-the-goal-process-works).

In particular, it would be good to focus on actionable goals we think are achievable (or where milestones are achievable) in a 6 month or 12 month basis.

Off the top of my head, I can think of the following examples I personally think could be good examples of goals:

* The "optimize for size" efforts, particularly around panic + formatting bloat
* Missing library components, like tools for working with mutable statics
* Work to get better CI coverage for embedded targets (compilation or runtime tests), including benchmarks for code size or memory usage
* Optimization of async layout

**Reply by adamgreig** [reply]
Perhaps "stabilise build-std" too?

**Reply by jamesmunns** [reply]
Also:

* Support for `#[ramfunc]` or some other tool that can help with loading code into RAM at startup, and ensure all dependent functions (like libcore items) are also present

**Reply by adamgreig** [reply]
(via @thejpster)

* Support for forcing a function to be panic-free
* further cortex-r examples, libraries, support

**Reply by adamgreig** [reply]
(via @jannic)

* resolve questions around MMIO access, e.g. with new mmio_read/write methods on pointers, with well documented interactions with the memory model, optimiser, etc

at the time it was suggested this could even be a method in a new crate, or in cortex-m/riscv crates, which could one day move to std. Open question from @Dirbaio about whether this permits the "can be re-ordered wrt other memory accesses" property that we'd ideally like to have for performance, though.

it was noted this week that perhaps `nomem` in the `asm()` call can allow the right reordering properties we'd want.

**Comment by adamhott** [comment]
I'd like 1 or 2 working group members to privately review my proposal for the Rust Foundation project. Need volunteers. Need to add as contributor reviewers to github repo.

**Comment by jannic** [comment]
Would you mind accepting me as a member of the triage team? The number of open tickets in many of our repos makes me think that it would be a useful way to help the WG.

**Reply by jannic** [reply]
PR: https://github.com/rust-embedded/wg/pull/779

**Comment by folkertdev** [comment]
mostly a PSA, but @tdittr and I have been working on moving the `cmse` features along, hopefully we can push for stabilization soon

- https://github.com/rust-lang/rust/pull/127814
- https://github.com/rust-lang/rust/pull/127766

I've also been trying to push naked functions forward

- https://github.com/rust-lang/rust/pull/127853
- https://github.com/rust-lang/rust/pull/128004

If you are/want to be using these features, we'd love to have your feedback
